docs/code.md

```diff
@@ -0,0 +1,25 @@
+# Code examples
+
+Giving a short explanation here for function:
+
+(add_function)=
+## Add Function
+
+```python
+def add(x, y):
```

Comment:
this code is not actually run yet, right? I could have syntactically nonsense code and that'd be fine?  Do we have a ticket for working out how to integrate quarto or similar?

Reply:
yep can make a ticket, so before doing this PR i had a play around with code executing packages, sphinx doctest unfortunately will not be good for us because while it does execute code, it does not attach the output to the next line. There is one other package I found called autorun and that is like having a python console and will have the output of the code on the next line but each code block is like a different python console so no setup code. I had a look at the autorun source code and it doesnt seem super difficult, I think we should just make our own code executing sphinx extension

Reply:
Did a quick demo, see #34 for a automatic code execution docs example and let me know what you think!